backend/controllers/followController.js

```diff
@@ -0,0 +1,154 @@
+const { User, ActivityLog, Follow } = require("../models");
+
+const putfollowAndUnfollow = async (req, res, next) => {
+  const { id } = req.params;
+  const user_id = req.user;
+  try {
+    const check = await Follow.findOne({
+      where: { follower_id: user_id, following_id: id },
+      order: [["createdAt", "DESC"]],
+    });
+
+    if (check && check.flag) {
+      await Follow.update({ flag: false }, { where: { id: check.id } });
+      await ActivityLog.create({
+        relatable_id: check.id,
+        relatable_type: "unfollows",
+        user_id,
+      });
+      res.status(200).json({ message: "Unfollow operation succeded" });
+    } else if (check && check.flag === false) {
+      await Follow.update({ flag: true }, { where: { id: check.id } });
+      await ActivityLog.create({
+        relatable_id: check.id,
+        relatable_type: "follows",
+        user_id,
+      });
+      res.status(200).json({ message: "Follow operation succeded" });
+    } else {
+      const follow = await Follow.create({
+        follower_id: user_id,
+        following_id: id,
+        flag: true,
+      });
+      await ActivityLog.create({
+        relatable_id: follow.id,
+        relatable_type: "follows",
+        user_id: user_id,
+      });
```

Comment:
- move this also to model class
- do the same to all other functions that has complex logic